template/src/ontology/Makefile.jinja2

```diff
@@ -814,26 +814,49 @@ $(ONT).owl: $(ONT)-{{ project.primary_release }}.owl
 
+ROBOT_RELEASE_IMPORT_MODE={% if project.use_edit_file_imports -%}$(ROBOT) merge --input $< {% else -%}
+$(ROBOT) remove --input $< --select imports --trim false {% if project.use_dosdps or project.components is defined -%}
+merge $(patsubst %, -i %, $(OTHER_SRC)) {% endif %}{% if project.import_group is defined -%}
+merge $(patsubst %, -i %, $(IMPORT_FILES)){% endif %}{% endif %}
+
+{% if 'base' in project.release_artefacts or project.primary_release == 'base' -%}
 # base: OTHER sources of interest, such as definitions owl
-$(ONT)-base.owl: $(EDIT_PREPROCESSED) $(OTHER_SRC)
+$(ONT)-base.owl: $(EDIT_PREPROCESSED) $(OTHER_SRC) $(IMPORT_FILES)
+	$(ROBOT_RELEASE_IMPORT_MODE) \{% if project.release_use_reasoner %}
+	reason --reasoner {{ project.reasoner }} --equivalent-classes-allowed {{ project.allow_equivalents }} --exclude-tautologies {{ project.exclude_tautologies }} \{% endif %}
+	relax \{% if project.release_use_reasoner %}
+	reduce -r {{ project.reasoner }} \{% endif %}
+	remove {% if project.namespaces is not none %}{% for iri in project.namespaces %}--base-iri {{iri}} {% endfor %}{% else %}--base-iri $(URIBASE)/{{ project.id.upper() }} {% endif %}--axioms external --preserve-structure false --trim false \
+	$(SHARED_ROBOT_COMMANDS) \
+	annotate --link-annotation http://purl.org/dc/elements/1.1/type http://purl.obolibrary.org/obo/IAO_8000001 \
+		--ontology-iri $(ONTBASE)/$@ $(ANNOTATE_ONTOLOGY_VERSION) \
+		{% if project.release_date -%} --annotation oboInOwl:date "$(OBODATE)" {% endif -%}--output $@.tmp.owl && mv $@.tmp.owl $@
+{% endif -%}
+
+{% if 'editors' in project.release_artefacts or project.primary_release == 'editors' -%}
+# base: OTHER sources of interest, such as definitions owl
+$(ONT)-editors.owl: $(EDIT_PREPROCESSED) $(OTHER_SRC)
 	$(ROBOT) remove --input $< --select imports --trim false \
-		{% if project.use_dosdps or project.components is defined %}merge $(patsubst %, -i %, $(OTHER_SRC)) \
-		{% endif %} $(SHARED_ROBOT_COMMANDS) annotate --link-annotation http://purl.org/dc/elements/1.1/type http://purl.obolibrary.org/obo/IAO_8000001 \
+		$(ROBOT_RELEASE_IMPORT_MODE) \
+		$(SHARED_ROBOT_COMMANDS) annotate --link-annotation http://purl.org/dc/elements/1.1/type http://purl.obolibrary.org/obo/IAO_8000001 \
 		--ontology-iri $(ONTBASE)/$@ $(ANNOTATE_ONTOLOGY_VERSION) \
 		{% if project.release_date -%} --annotation oboInOwl:date "$(OBODATE)" {% endif -%}--output $@.tmp.owl && mv $@.tmp.owl $@
+{% endif -%}
 
+{% if 'full' in project.release_artefacts or project.primary_release == 'full' -%}
 # Full: The full artefacts with imports merged, reasoned
 $(ONT)-full.owl: $(EDIT_PREPROCESSED) $(OTHER_SRC) $(IMPORT_FILES)
-	$(ROBOT) merge --input $< \
-		reason --reasoner {{ project.reasoner }} --equivalent-classes-allowed {{ project.allow_equivalents }} --exclude-tautologies {{ project.exclude_tautologies }} \
-		relax \
-		reduce -r {{ project.reasoner }} \
+	$(ROBOT_RELEASE_IMPORT_MODE) \{% if project.release_use_reasoner %}
```

Comment:
@gouttegd warns about confusing definition of full.

Other option introduce alternative release artefacts: 
- add full-non-reasoned.

At least, based on the name of the product, you will know exactly the specification of the release, there is no doubt.

@matentzn things this is a good idea and will contemplate it.

Reply:
Random thoughts: What if we generalise the concept by having release products follow a strict naming scheme, made of `-`-separated components where each component indicate a feature of the release (for example, whether reasoning has been performed or not, and whether imports have been merged or not).

We could have a naming scheme of `$(ont)-$(imports)-$(reasoning)`, which would give for example:

* `fbbt-imports-reasoned.owl` for a version of FBbt with imports merged and inferred axioms;
* `fbbt-noimports-unreasoned.owl` for a version of FBbt without imports nor inferred axioms;
* etc.

Then for backwards compatibility, we could keep (at least temporarily) the old names of `$(ont)-base`, `$(ont)-full`, etc. but just aliasing them to the appropriate “canonical” name (the name that follows the above naming scheme).

Individual ontologies would still have the possibility of enabling/disabling some products (not everybody will want/need to produce all possible combinations of features), and maybe we could even allow an ontology to specify their own aliasing for the `$(ont).owl` product (by default this could be `$(ont)-imports-reasoned.owl`, but an ontology could redefine it to be `$(ont)-imports-unreasoned.owl` for example).

The scheme can be extended to more than 3 components (base name, imports feature, reasoning feature) if needed.

Reply:
Definitely worth considering.

Reply:
For this version ODK, I retracted my changes to leave the canonical "full" as is.